pkg/apis/kops/cluster.go

```diff
@@ -160,6 +160,8 @@ type ClusterSpec struct {
 	IAM *IAMSpec `json:"iam,omitempty"`
 	// EncryptionConfig controls if encryption is enabled
 	EncryptionConfig *bool `json:"encryptionConfig,omitempty"`
+	// TerraformProviderExtraConfig contains key/value pairs to add to the rendered terraform "provider" block
+	TerraformProviderExtraConfig map[string]string `json:"terraformProviderExtraConfig,omitempty"`
```

Comment:
I'm wondering if we might want to do this slightly differently to allow changes like these for the other targets and tf features down the road.  Rather than defining `TerraformProviderExtraConfig` maybe we should structure it more? That way config is something like:

```
target:
  terraform:
    providerExtraConfig:
      alias: foo
```

This may seem like too much for now but I'm concerned about creating this as a one off since we'll likely want to allow other tf changes in the future like version locking in the provider:

```
target:
  terraform:
    providerVersion: "~> 1.0"
```